agents/tests/test_dns_agent.py

```diff
@@ -0,0 +1,163 @@
+# Command to run on VM: python -m pytest agents/tests/test_dns_agent.py
+import os
+import tempfile
+
+import pytest
+from mock import MagicMock, patch
+
+from agents.dns.dns import DNSAgent
+
+
+@pytest.yield_fixture
+def dns_agent():
+    """
+    Create and configure a DNSAgent instance with logging for use in tests.
+    Cleans up the temporary log file after the test completes.
+    """
+    os.environ['PORT'] = '53'
+    logfile = tempfile.NamedTemporaryFile(delete=False)
+    logfile.close()
+
+    agent = DNSAgent(query_domain='google.com')
+    agent.setup_logging(logfile.name)
+
+    yield agent
+
+    if os.path.exists(logfile.name):
+        os.remove(logfile.name)
+
+
+def test_status_to_dict_keys(dns_agent):
+    """
+    Verify that status_to_dict() returns all expected keys
+    in the status dictionary.
+    Uses mock to isolate from metadata collection implementation.
+    """
+    with patch.object(dns_agent, 'collect_server_metadata') as mock_collect:
+        dns_agent.os_type = 'linux'
+        dns_agent.hostname = 'test-host'
+        dns_agent.ip = '127.0.0.1'
+        dns_agent.server_name = 'dns'
+        dns_agent.uptime = 12345
+        dns_agent.timestamp = '2025-06-03 20:00:00'
+        dns_agent.healthy = True
+
+        result = dns_agent.status_to_dict()
+
+        required_keys = set([
+            'os',
+            'hostname',
+            'ip',
+            'server_name',
+            'uptime',
+            'timestamp',
+            'healthy',
+            ])
+        mock_collect.assert_called_once()
+
+        assert set(result.keys()) == required_keys
+
+
+def test_status_to_dict_with_missing_fields(dns_agent):
+    """
+    Ensure status_to_dict() handles missing or None fields gracefully.
+    """
+    with patch.object(dns_agent, 'collect_server_metadata'):
+        dns_agent.os_type = None
+        dns_agent.hostname = None
+        dns_agent.ip = None
+        dns_agent.server_name = 'dns'
+        dns_agent.uptime = -1
+        dns_agent.timestamp = None
+        dns_agent.healthy = False
+
+        result = dns_agent.status_to_dict()
+
+        assert result['os'] is None
+        assert result['hostname'] is None
+        assert result['ip'] is None
+        assert result['uptime'] == -1
+        assert result['healthy'] is False
```

Comment:
These two tests don't test DNS-specific logic. But perhaps it's OK like this for now - we'll decide how to unify this check in `test_agent.py` later on.